drivers/tee/optee/optee.c

```diff
@@ -1230,6 +1268,16 @@ static int optee_init(const struct device *dev)
 		return -ENOTSUP;
 	}
 
+	/*
+	 * We need to disable all existing shm objects before enabling caching.
+	 * This prevents us from receiving invalid shm addresses.
+	 */
+	optee_disable_shm_cache(dev, false);
+
+	if (!data->rpc_param_count) {
```

Comment:
Why you are enabling cache only if `rpc_param_count==0`? I see no such check in the linux kernel driver